parser/include/parser/abstractparser.h

```diff
@@ -33,6 +33,8 @@ class AbstractParser
    */
   virtual std::vector<std::string> getDependentParsers() const = 0;
 
+  virtual void preparse(){}
```

Comment:
A documentation for this function should be given (and I agree `parse()`'s one is lacking too...). I'm not too confident a `void` return value is a good choice here (perhaps `bool` could indicate that the subclass parser failed to maintain its datasets and thus should not be called --- perhaps add this as a TODO).

Also, the default implementation of just *do nothing* feels weird.

Reply:
Good point, implemented in 7100014093aed0a1ae46a02ef6c1bff82a641139.